storage/file_persister_test.go

```diff
@@ -74,3 +79,174 @@ func TestLocalFilePersister(t *testing.T) {
 		})
 	}
 }
+
+func TestRemoteFilePersister(t *testing.T) {
+	t.Parallel()
+
+	basePath := "screenshots"
+
+	tests := []struct {
+		name                    string
+		path                    string
+		data                    string
+		wantPresignedURLBody    string
+		wantPresignedHeaders    map[string]string
+		uploadResponse          int
+		getPresignedURLResponse int
+		wantError               string
+	}{
+		{
+			name: "upload_file",
+			path: "some/path/file.png",
+			data: "here's some data",
+			wantPresignedURLBody: `{
+					"service":"aws_s3",
+					"files":[{"name":"screenshots/some/path/file.png"}]
+				}`,
+			wantPresignedHeaders: map[string]string{
+				"Authorization": "token asd123",
+				"Run_id":        "123456",
+			},
+			uploadResponse:          http.StatusOK,
+			getPresignedURLResponse: http.StatusOK,
+		},
+		{
+			name: "get_presigned_rate_limited",
+			path: "some/path/file.png",
+			data: "here's some data",
+			wantPresignedURLBody: `{
+					"service":"aws_s3",
+					"files":[{"name":"screenshots/some/path/file.png"}]
+				}`,
+			wantPresignedHeaders: map[string]string{
+				"Authorization": "token asd123",
+				"Run_id":        "123456",
+			},
+			getPresignedURLResponse: http.StatusTooManyRequests,
+			wantError:               "get presigned url: status code out of range: request is being rate limited",
```

Comment:
I believe we shouldn't do exact matches for errors for refactorability later on. Do you think just checking "request is being rate limited" is fine here?

Reply:
Agreed that this is tightly coupling the implementation and the test. I actually found this quite useful to test how the errors would look so I can shape them better. Happy to change them though if you have a strong opinion on it.

Reply:
I'm leaning on the side of them being helpful otherwise the error comparison for both the presigned and upload responses would read the same `request is being rate limited`, whereas the full error message ensures that the correct paths within the implementation have been taken.